include/highfive/H5Easy.hpp

```diff
@@ -0,0 +1,217 @@
+/*
+ *  Copyright (c), 2017, Adrien Devresse <adrien.devresse@epfl.ch>
+ *
+ *  Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ *
+ */
+#ifndef H5EASY_HPP
+#define H5EASY_HPP
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "H5File.hpp"
+#include "H5DataSet.hpp"
+#include "H5DataType.hpp"
+#include "H5DataSpace.hpp"
+
+// optionally enable plug-in xtensor and load the library
+#ifdef XTENSOR_VERSION_MAJOR
+#define HIGHFIVE_XTENSOR
+#endif
+
+#ifdef HIGHFIVE_XTENSOR
+#include <xtensor/xarray.hpp>
+#include <xtensor/xtensor.hpp>
+#endif
+
+// optionally enable plug-in Eigen and load the library
+#ifdef EIGEN_WORLD_VERSION
+#define HIGHFIVE_EIGEN
+#endif
+
+#ifdef HIGHFIVE_EIGEN
+#include <Eigen/Eigen>
+#endif
+
+namespace HighFive
+{
+    enum Mode
+    {
+      Create,
+      Overwrite
+    };
+
+    ///
+    /// Get the groupName of a path to a DataSet. For example for ``path = "/path/to/dataset"`` this
+    /// function returns ``"/path/to"``.
+    ///
+    /// @param path path to a DataSet
+    ///
+    /// @return group the path of the group above the DataSet
+    inline std::string groupName(const std::string& path);
+
+    ///
+    /// Recursively create groups in an open HDF5 file such that a DataSet can be created.
+    /// For example for ``path = "/path/to/dataset"``, this function will create the groups
+    /// ``"/path"`` and ``"/path/to"``.
+    ///
+    /// @param file opened HighFive::File
+    /// @param path path of the DataSet
+    ///
+    inline void createGroupsToDataSet(HighFive::File& file, const std::string& path);
+
+    ///
+    /// Get the size of an existing DataSet in an open HDF5 file.
+    ///
+    /// @param file opened HighFive::File
+    /// @param path path of the DataSet
+    ///
+    /// @return size the size of the HighFive::DataSet
+    inline auto size(const HighFive::File& file, const std::string& path);
```

Comment:
HighFive is C++11 compatible. I am afraid it does not support `auto` return type.

Reply:
...

That's really a pity.